### PR TITLE
feat: Add AssociatedResources field to Manifest Spec

### PR DESCRIPTION
**Pull request by nesmabadr**
**Description**

Changes proposed in this pull request:

- Add `associatedResources` to manifest.spec
- Add documentation

**Related issue(s)**
#1582

### CODE REVIEW

File: api/v1beta2/manifest_types.go [hunk 1]
```diff
@@ -62,6 +62,10 @@ type ManifestSpec struct {
 	// +nullable
 	// Resource specifies a resource to be watched for state updates
 	Resource *unstructured.Unstructured `json:"resource,omitempty"`
+
+	// ManagedResources specifies a list of resources which are managed
+	// manually by the user. The resources have the following format: <group>/<version>/<kind>.
+	ManagedResources []string `json:"managedResources,omitempty"`
```
Comment (Tomasz-Smelcerz-SAP) [inline api/v1beta2/manifest_types.go:68]: How about adding `+optional` marker? Maybe it is not strictly required, but we use it already in line 51 for example.

File: docs/technical-reference/api/manifest-cr.md [hunk 2]
```diff
@@ -85,6 +85,17 @@ The [internal spec resolver](../../../internal/manifest/spec_resolver.go) uses t
 
 The resource is the default data that should be initialized for the module and is directly copied from **.spec.data** of the ModuleTemplate CR after normalizing it with the **namespace** for the synchronized module.
 
+### **.spec.managedResources**
+
+The managedResources are the resources which are managed manually by the user. They are represented by the `group/version/kind` format. 
+If the module is managed and gets deleted from the Kyma CR, module resources remain undeleted until all custom resources with GVK listed in the spec.managedResources are removed manually by the user.
```
Comment (Tomasz-Smelcerz-SAP) [inline docs/technical-reference/api/manifest-cr.md:91]:
```suggestion
If the module is managed and gets deleted from the Kyma CR, module deletion is suspended until all custom resources with GVK listed in the spec.managedResources are removed manually by the user.   
```

File: api/v1beta2/manifest_types.go (hunk 1 above)
Comment (nesmabadr) [reply api/v1beta2/manifest_types.go:68]: I can add it but it is the same as omit empty

**Review by Tomasz-Smelcerz-SAP — DISMISSED** [review]

File: docs/technical-reference/api/manifest-cr.md [hunk 3]
```diff
@@ -85,6 +85,17 @@ The [internal spec resolver](../../../internal/manifest/spec_resolver.go) uses t
 
 The resource is the default data that should be initialized for the module and is directly copied from **.spec.data** of the ModuleTemplate CR after normalizing it with the **namespace** for the synchronized module.
 
+### **.spec.managedResources**
+
+The managedResources are the resources which are managed manually by the user. They are represented by the `group/version/kind` format.
```
Comment (mmitoraj) [inline docs/technical-reference/api/manifest-cr.md:90]:
```suggestion
The managed resources are the resources that are handled manually by the user. They are represented by the `group/version/kind` format. 
```

File: docs/technical-reference/api/manifest-cr.md [hunk 4]
```diff
@@ -85,6 +85,17 @@ The [internal spec resolver](../../../internal/manifest/spec_resolver.go) uses t
 
 The resource is the default data that should be initialized for the module and is directly copied from **.spec.data** of the ModuleTemplate CR after normalizing it with the **namespace** for the synchronized module.
 
+### **.spec.managedResources**
+
+The managedResources are the resources which are managed manually by the user. They are represented by the `group/version/kind` format. 
+If the module is managed and gets deleted from the Kyma CR, module deletion is suspended until all custom resources with GVK listed in the spec.managedResources are removed manually by the user.
```
Comment (mmitoraj) [inline docs/technical-reference/api/manifest-cr.md:91]:
```suggestion
Suppose the module is managed and gets deleted from the Kyma CR. In that case, the module deletion is suspended until all custom resources with GVK listed in the **spec.managedResources** are removed manually by the user.
```

**Review by mmitoraj — COMMENTED** [review]
As agreed with Nesma, let's discuss the naming with the team. In that description, "managed" is used in the context of modules and resources. In the context of modules, "managed" means handled by the Kyma team which implicates some kind of automation for the user. In the context of resources, "managed" means handled by the user, which requires some manual action at their end. For me, it is confusing. Instead of "managedResources" maybe we could say "userResources" but I'm not an expert. :)

**Review by mmitoraj — DISMISSED** [review]

File: docs/technical-reference/api/manifest-cr.md [hunk 5]
```diff
@@ -85,6 +85,17 @@ The [internal spec resolver](../../../internal/manifest/spec_resolver.go) uses t
 
 The resource is the default data that should be initialized for the module and is directly copied from **.spec.data** of the ModuleTemplate CR after normalizing it with the **namespace** for the synchronized module.
 
+### **.spec.associatedResources**
+
+The associated resources are the resources that are not managed by the lifecycle manager. They are represented by the `group/version/kind` format.
```
Comment (mmitoraj) [inline docs/technical-reference/api/manifest-cr.md:90]:
```suggestion
Associated resources are resources that are not managed by Lifecycle Manager. They are represented by the `group/version/kind` format.  
```

File: docs/technical-reference/api/manifest-cr.md [hunk 6]
```diff
@@ -85,6 +85,17 @@ The [internal spec resolver](../../../internal/manifest/spec_resolver.go) uses t
 
 The resource is the default data that should be initialized for the module and is directly copied from **.spec.data** of the ModuleTemplate CR after normalizing it with the **namespace** for the synchronized module.
 
+### **.spec.associatedResources**
+
+The associated resources are the resources that are not managed by the lifecycle manager. They are represented by the `group/version/kind` format.  
+Suppose the module is managed and gets deleted from the Kyma CR. In that case, the module deletion is suspended until all custom resources with GVK listed in the **spec.managedResources** are removed manually by the user.
```
Comment (mmitoraj) [inline docs/technical-reference/api/manifest-cr.md:91]:
```suggestion
Suppose the module is managed and gets deleted from the Kyma CR. In that case, the module deletion is suspended until all custom resources with GVK listed in the **spec.associatedResources** are removed manually by the user.   
```

File: api/v1beta2/manifest_types.go [hunk 7]
```diff
@@ -62,6 +62,11 @@ type ManifestSpec struct {
 	// +nullable
 	// Resource specifies a resource to be watched for state updates
 	Resource *unstructured.Unstructured `json:"resource,omitempty"`
+
+	// AssociatedResources specifies a list of resources which are not managed
+	// by the lifecycle manager. The resources have the following format: <group>/<version>/<kind>.
```
Comment (mmitoraj) [inline api/v1beta2/manifest_types.go:67]:
```suggestion
	// by Lifecycle Manager. The resources have the following format: <group>/<version>/<kind>.
```

File: config/crd/bases/operator.kyma-project.io_manifests.yaml [hunk 8]
```diff
@@ -49,6 +49,13 @@ spec:
           spec:
             description: ManifestSpec defines the desired state of Manifest.
             properties:
+              associatedResources:
+                description: |-
+                  AssociatedResources specifies a list of resources which are not managed
+                  by the lifecycle manager. The resources have the following format: <group>/<version>/<kind>.
```
Comment (mmitoraj) [inline config/crd/bases/operator.kyma-project.io_manifests.yaml:55]:
```suggestion
                  by Lifecycle Manager. The resources have the following format: <group>/<version>/<kind>.
```

**Review by mmitoraj — APPROVED** [review]

**Review by Tomasz-Smelcerz-SAP — APPROVED** [review]

**Pull request merged**